packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -464,6 +465,8 @@ export const prepareAuctionBook = (baggage, zcf, makeRecorderKit) => {
             AmountMath.make(collateralBrand, QUOTE_SCALE),
             bidBrand,
           );
+          baggage.set(QUOTE_NOTIFIER, quoteNotifier);
```

Comment:
I think that's storing a promise in (durable) baggage, which isn't allowed (also that `const quoteNotifier =` should probably be `quoteNotifierP =` to make that more obvious). Would it hurt anything to `await` and store the actual Presence instead?

Reply:
No.

 I also notice that we don't seem to make a new baggage for each book. I need to figure out if that's true, and either make new baggages, or make unique names for each notifier

Reply:
Oh, no, that wouldn't be a trivial change. It would make `observeQuoteNotifier` async, but also it wouldn't be setting the flag at the right time: if two callers invoke `observeQuoteNotifier` in the same turn, we'd still wind up with multiple outstanding calls to `makeQuoteNotifier`, and we'd get multiple observers again.

I think `observerQuoteNotifier()` needs to be named/intentioned "maybeObserveQuoteNotifier" or "maybeStartObservingQuoteNotifier". And it needs to be "async idempotent" (not sure if that's a term of art but you get the idea): no matter how many times you call it in a row, you still only wind up with a single outstanding async request, but if that request fails for any reason, it resets back to a state where you can call it again and it won't short-circuit.

The flag needs to be checked on entry to the function, during the synchronous prelude, before sending any messages. Then it needs to send off the `makeQuoteNotifier`, and if that fails it needs to clear the flag. If it succeeds then it can build the observer, which clears the flag if/when the observer fails or finishes.

Reply:
updated via pairing.